From fc3beaeb683747ca5aa9f26654cb38e1de3f2d84 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Fri, 11 Jan 2019 15:43:44 +0000
Subject: [PATCH 747/909] media: i2c: adv7482: Fix wait procedure usleep_range
 from msleep

commit 0786087e9766327cc9f741013a3f54de87783bb3 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

By Documentation/timers/timers-howto.txt, when waiting 20ms from 10us,
it is correct to use usleep_range. this patch corrects it.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
(cherry picked from horms/renesas-bsp commit af0cdba377bc
("media: i2c: adv7482: Fix wait procedure usleep_range from msleep"))
Signed-off-by: Steve Longerbeam <steve_longerbeam@mentor.com>
Link: https://patchwork.kernel.org/patch/10758319/
[koji.matsuoka.xm: the part of the patch for kernel v4.14.75
was rebased and imported]
Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/i2c/adv748x/adv748x-core.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/adv748x/adv748x-core.c b/drivers/media/i2c/adv748x/adv748x-core.c
index a8cdf3f44123..cd1fc6c29b21 100644
--- a/drivers/media/i2c/adv748x/adv748x-core.c
+++ b/drivers/media/i2c/adv748x/adv748x-core.c
@@ -1,7 +1,7 @@
 /*
  * Driver for Analog Devices ADV748X HDMI receiver with AFE
  *
- * Copyright (C) 2017 Renesas Electronics Corp.
+ * Copyright (C) 2017-2019 Renesas Electronics Corp.
  *
  * This program is free software; you can redistribute  it and/or modify it
  * under  the terms of  the GNU General  Public License as published by the
@@ -214,7 +214,8 @@ static int adv748x_write_regs(struct adv748x_state *state,
 
 	while (regs->page != ADV748X_PAGE_EOR) {
 		if (regs->page == ADV748X_PAGE_WAIT) {
-			msleep(regs->value);
+			usleep_range(regs->value * 1000,
+				     (regs->value * 1000) + 1000);
 		} else {
 			ret = adv748x_write(state, regs->page, regs->reg,
 				      regs->value);
-- 
2.17.1

